package: langchain-voyageai

fzowl opened the pull request:
Thank you for contributing to LangChain!

**Description:** 
Introducing VoyageAI's new embedding and rerank models


- [ ] **Lint and test**: Run `make format`, `make lint` and `make test` from the root of the package(s) you've modified. See contribution guidelines for more: https://python.langchain.com/docs/contributing/

thomas0809 commented on libs/community/langchain_community/embeddings/voyageai.py:
```diff
@@ -69,12 +69,12 @@ class VoyageEmbeddings(BaseModel, Embeddings):
+    model: str
     voyage_api_base: str = "https://api.voyageai.com/v1/embeddings"
     voyage_api_key: Optional[SecretStr] = None
     batch_size: int = 8
```
Please remove this line as it is no longer used.

thomas0809 replied:
This is not resolved yet.

fzowl replied:
Yes, makes sense. Made the change.

thomas0809 commented on docs/docs/integrations/text_embedding/voyageai.ipynb:
```diff
@@ -37,7 +37,7 @@
    "outputs": [],
    "source": [
-    "embeddings = VoyageEmbeddings(voyage_api_key=\"[ Your Voyage API key ]\")"
+    "embeddings = VoyageEmbeddings(voyage_api_key=\"[ Your Voyage API key ]\", model=\"[model name]\")"
```
Let's just use `model="voyage-2"` here. Please also add a sentence that directs to our docs, like the user can find all the models there https://docs.voyageai.com/embeddings/

thomas0809 reviewed (changes requested):
Please see the additional comments. Thanks!

thomas0809 commented on libs/community/langchain_community/embeddings/voyageai.py:
```diff
@@ -69,12 +69,11 @@ class VoyageEmbeddings(BaseModel, Embeddings):
             query_result = voyage.embed_query(text)
     """
 
-    model: str = "voyage-01"
```
It shouldn't be deleting this line, right? I think it should be just `model: str`.
Can it pass the test if you delete this line? I am not sure, actually.

fzowl replied:
Yes, `model: str` is a must. Sorry, i blindly removed this line because of your previous comment to remove unused attributes. (also, i need to switch back to Python from JavaScript/TypeScript)

thomas0809 reviewed (changes requested):
Please see the comments

fzowl committed:
VoyageAI Embedding provider
Adding voyage-3, voyage-3-lite, rerank-2 and rerank-2-lite to docs
Modify the default batch size

fzowl committed:
Reformatting

YuxingZhang commented on libs/partners/voyageai/langchain_voyageai/embeddings.py:
```diff
@@ -55,7 +55,15 @@ def default_values(cls, values: dict) -> Any:
         batch_size = values.get("batch_size")
         if batch_size is None:
-            values["batch_size"] = 72 if model in ["voyage-2", "voyage-02"] else 7
+            values["batch_size"] = (
+                72
+                if model in ["voyage-2", "voyage-02"]
+                else (
+                    30
+                    if model == "voyage-3-lite"
+                    else (10 if model == "voyage-3" else 7)
+                )
+            )
```
How did we get these numbers? @fzowl

YuxingZhang reviewed (approved):
LGTM besides my question. Please also wait for Hong to approve it before merging.

Liuhong99 commented:
lgtm. please go ahead and merge it. also please wait until both new rerankers and embeddings are officially online and then raise a pr to langchain
